ui/src/state/mod.rs

```diff
@@ -726,31 +719,41 @@ impl State {
 
     /// Saves the current state to disk.
     fn save(&self) -> Result<(), Box<dyn std::error::Error>> {
-        // mock data loads the same random data every time. no point in saving it.
-        if STATIC_ARGS.use_mock {
-            return Ok(());
-        }
         let serialized = serde_json::to_string(self)?;
-        fs::write(&STATIC_ARGS.cache_path, serialized)?;
+        let path = if STATIC_ARGS.use_mock {
+            &STATIC_ARGS.mock_cache_path
+        } else {
+            &STATIC_ARGS.cache_path
+        };
+        fs::write(path, serialized)?;
         Ok(())
     }
 
     /// Loads the state from a file on disk, if it exists.
-    pub fn load() -> Result<Self, std::io::Error> {
+    pub fn load() -> Self {
+        if STATIC_ARGS.use_mock {
+            return State::load_mock();
+        };
+
         let contents = match fs::read_to_string(&STATIC_ARGS.cache_path) {
             Ok(r) => r,
-            Err(_) => return Ok(State::default()),
+            Err(_) => {
+                return State::default();
+            }
         };
-        let state: State = serde_json::from_str(&contents).unwrap_or_default();
-        Ok(state)
+        serde_json::from_str(&contents).unwrap_or_default()
     }
 
-    pub fn mock() -> Self {
-        generate_mock()
+    fn load_mock() -> Self {
+        let contents = match fs::read_to_string(&STATIC_ARGS.mock_cache_path) {
+            Ok(r) => r,
+            Err(_) => {
+                return generate_mock();
+            }
+        };
+        serde_json::from_str(&contents).unwrap_or_else(|_| generate_mock())
     }
-}
 
```

Comment:
looks like commit `5ac67e4adee9171ee682caa40b4f0952ef31294e` removed the handling for `MessageEvent::ReactionAdded` and `MessageEvent::ReactionRemoved`

Reply:
and merging with dev fixed it. this one is resolved.